firestore/google/cloud/firestore_v1beta1/query.py

```diff
@@ -530,13 +533,48 @@ def _filters_pb(self):
             return query_pb2.StructuredQuery.Filter(
                 composite_filter=composite_filter)
 
+    @staticmethod
+    def _normalize_cursor(cursor, orders):
+        """Helper: convert cursor to a list of values based on orders."""
+        if cursor is None:
+            return
+
+        if not orders:
+            raise ValueError(_NO_ORDERS_FOR_CURSOR)
+
+        document_fields, before = cursor
+
+        order_keys = [order.field.field_path for order in orders]
+
+        if isinstance(document_fields, dict):
+            # Transform to list using orders
+            values = []
+            data = document_fields
+            for order_key in order_keys:
+                try:
+                    values.append(_helpers.get_nested_value(order_key, data))
```

Comment:
May be worth reformatting as the following as append is likely more expensive than a listcomp.
```
try:
    values = [_helpers.get_nested_value(order_key, data) for order_key in order_keys]
except:
```